executor/executor.yml

```diff
@@ -136,4 +136,50 @@
   delay: 1
   retries: 300
 
+- name: Fetching verdict from the chaos result CR
+  shell: >
+     kubectl get chaosresult -n {{ c_app_ns }} -o jsonpath='{.items[?(@.metadata.name=="{{ c_engine }}-{{ c_experiment_name }}")].spec.experimentstatus.verdict}'
+  args:
+   executable: /bin/bash
+  register: c_verdict
 
+- name: Fetching phase from the chaos result CR
+  shell: >
+     kubectl get chaosresult -n {{ c_app_ns }} -o jsonpath='{.items[?(@.metadata.name=="{{ c_engine }}-{{ c_experiment_name }}")].spec.experimentstatus.phase}'
+  args:
+   executable: /bin/bash
+  register: c_phase
+
+- debug:
+    msg: "verdict : {{c_verdict.stdout}}"
+
+- name: Fetching chaos engine yaml file
+  shell: >
+    kubectl get chaosengine {{c_engine}} -n {{ c_app_ns }} -oyaml > chaos-engine.yaml
+  args:
+    executable: /bin/bash
+
+- name: patch the chaosengine status
+  lineinfile:
+    dest: chaos-engine.yaml
+    state: present
+    line: '{{item}}'
+  with_items:
+  - "  - name: {{c_experiment_name}}"
+  - "    status: {{c_phase.stdout}}"
+  - "    verdict: {{c_verdict.stdout}}"
```

Comment:
Can you show an output of how the patched engine looks? The expectation is similar:

```yaml
status:
  experimentStatuses:
     - name:  {{c_experiment_name}}
        instance: {{c_experiment_job_name}} #we can derive this, helps during scheduled chaos
        status: {{c_phase.stdout}}
        verdict: {{c_verdict.stdout}}
```

Reply:
Also, would it help to have a jinja template with the spec that we can just append using blockinfile?

Reply:
Also, was this tested with the chaosengine CRD having status as subresource? If this is removed, will the limitation to set an empty experiment list in status be solved?

Reply:
> Also, would it help to have a jinja template with the spec that we can just append using blockinfile?

In this approach we have to handle all the dynamic value in jinja template .It will be complex if the env variable are present in engine spec. In that case we have to modify template accordingly

Reply:
> Can you show an output of how the patched engine looks? The expectation is similar to below:
> 
> ```yaml
> status:
>   experimentStatuses:
>   - name:  {{c_experiment_name}}
>     instance: {{c_experiment_job_name}} #we can derive this, helps during scheduled chaos
>     status: {{c_phase.stdout}}
>     verdict: {{c_verdict.stdout}}
> ```

Limitation of having empty list can we remove if we use status.experimentStatuses. Initially I was using status.experiments .It has conflict with spec.experiments and blockinfile didn't work as expected due to that conflict